### DISCUSSION TITLE
repeat until and repeat option pair in code editor block right click menu (with a certain addon)

### DISCUSSION THREAD

**Discussion by ksdavidc** [ScratchAddons/ScratchAddons#4677] [Ideas]
I don't know the correct name of the particular addon/setting, but when right-clicking on a block in the right editor, this addon/setting gives you other options (e.g., right-clicking a * block shows +. ÷, - as options). I suggest making “repeat” and “repeat until” be options (for each other) (when that particular addon/setting is enabled).

**Comment by Endermaster3690** [comment]
what if someone has a bool value and switches it from "repeat until <>" to "repeat ()"

**Reply by mxmou** [reply]
This has been suggested before and closed for this exact reason: #1107

It might not actually be a bad idea though because replacing a c-block while keeping its contents is currently difficult. Of course it doesn't make sense to keep the repeat count/condition.

**Reply by ksdavidc** [reply]
I understand the problem and can propose a solution, but I imagine there might be several ways to address it, or also that I am missing some important/obvious reason why it can't work that way:

As I understand it, if you switch from say a boolean (repeat until + boolean) to an integer (repeat + integer (rounded up real)) there is a problem of type. But at the same time, booleans are sometimes interpretable as  integers in the set {0,1}. So, we could imagine that happening in the resultant repeat () block,....

except, what does repeat 0 or repeat 1 mean? They don't break the code per se as the type is alright, but the "natural meaning" (skip, do once) are arguable redundant/useless. 

So I propose just emptying the field, and taking the contents of the boolean and placing it nearby as a new block. This does not lose information, as there may be some information (aka a variable, etc.) in the original block that can be of service in the new. Meanwhile it would avoid the problem before. 

The reverse conversion, from repeat () to repeat until () is not as neat, since a number isn't a boolean. It might work if the parameter of the repeat is a variable say, but that is not always the case. One way to address that would be to have a "default" boolean, such as  < () = () >  , and put whatever was in the original repeat in for the second numerical parameter. This would preserve information, but allow conversion.

I would be happy enough with that. Of course I am probably missing something, but that is how I would envision it working.
